patches/server/0882-Added-EntityToggleSitEvent.patch

```diff
@@ -0,0 +1,48 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: KyGuy2002 <IEatBeans#1165>
+Date: Fri, 11 Mar 2022 15:33:10 +0000
+Subject: [PATCH] Added EntityToggleSitEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/TamableAnimal.java b/src/main/java/net/minecraft/world/entity/TamableAnimal.java
+index 7e67c47434d1f29cc87f3af926f3efc920c87829..b4a958f0a7380772f4b0d71a1aef3b8fa3fb8157 100644
+--- a/src/main/java/net/minecraft/world/entity/TamableAnimal.java
++++ b/src/main/java/net/minecraft/world/entity/TamableAnimal.java
+@@ -126,6 +126,12 @@ public abstract class TamableAnimal extends Animal implements OwnableEntity {
+     }
+ 
+     public void setInSittingPose(boolean inSittingPose) {
++        // Paper start - call EntityToggleSitEvent
++        io.papermc.paper.event.entity.EntityToggleSitEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityToggleSitEvent(this, inSittingPose);
++        if (event.isCancelled()) {
```

Comment:
construct the event, and call it right in the same line, no need for a local var for the event.

Reply:
I dont think im able to do this if i call the event right there as well, since the .callEvent doesnt return anything so i cannot check if its cancellled.  I can call it there but i belive i will still need a local var?  Thanks

Reply:
callEvent returns a boolean based on if it was cancelled or not.

Reply:
![image](https://user-images.githubusercontent.com/75859301/157922851-633c84ff-614b-4258-8f1b-f5d4951016b6.png)
Sorry if im not getting you, but it doesnt appear that thats the case?  Im using `Bukkit.getPluginManager().callEvent(Event)`

Reply:
no, callEvent on the Event object.
`new Event(whatever).callEvent()`

Reply:
oh alright, thanks